static/js/publisher/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+
+function useLocalStorage<T = unknown>(
+  key: string,
+  initialValue?: T | (() => T),
+): [T, (value: T | ((prevValue: T) => T)) => void] {
+  // use a function to initialize the state to ensure init only runs on first render
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    const item = window.localStorage.getItem(key);
+    return item ? JSON.parse(item) : initialValue;
+  });
+
+  // when state changes update local storage
+  useEffect(() => {
+    setTimeout(() => {
+      window.localStorage.setItem(key, JSON.stringify(storedValue));
+    }); // delay the write to let React do its job first
```

Comment:
I think it's good to but a numeric value of `0` here to make it clear that there is no time length intentionally